File: src/main/org/nlogo/nvm/Tracer.java
package org.nlogo.nvm;

public abstract strictfp class Tracer {

  // Tracer control interface
  public abstract void enable();

  public abstract void disable();

  public abstract void reset();

  public abstract void dump(java.io.PrintStream stream);

  // Tracer call recording interface
  public abstract void openCallRecord(Context context, Activation activation);

  public abstract void closeCallRecord(Context context, Activation activation);

  public abstract long calls(String name);

  public abstract long exclusiveTime(String name);

  public abstract long inclusiveTime(String name);
}

File: netlogo-core/src/main/nvm/Tracer.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.nvm

import java.util.Set

abstract class Tracer {

  // control
  def enable(): Unit
  def disable(): Unit
  def reset(): Unit
  def dump(stream: java.io.PrintStream): Unit

  // call recording
  def openCallRecord(context: Context, activation: Activation): Unit
  def closeCallRecord(context: Context, activation: Activation): Unit
  def calls(name: String): Long
  def exclusiveTime(name: String): Long
  def inclusiveTime(name: String): Long

  def procedureNames(): Set[String]

}
